src/state/nftMarket/helpers.ts

```diff
@@ -1024,7 +1155,7 @@ export const getCompleteAccountNftData = async (
     })
     .map((nft) => nft.tokenId)
 
-  const marketDataForSaleNfts = await getNftsMarketData({ currentSeller: account.toLowerCase() })
+  const marketDataForSaleNfts = await fetchMarketDataForSaleNfts(collections, account)
```

Comment:
Can we do this ?

```
const marketDataForSaleNfts = await getNftsMarketData({ currentSeller: account.toLowerCase() })

const tokenIdsForSale = marketDataForSaleNfts.map((nft) => nft.tokenId)

const onChainMarketDataForSaleNfts = await onChainFetch(tokenIdsForSale)

return {...marketDataForSaleNfts, ...onChainMarketDataForSaleNfts}
```

Reply:
We should first get the tokenIds from onchain not from graph api. Actually currently onchain is enough since we only need seller and price data but i added to fetch from graph api to fill the other data (tradeVolumeBNB, metadataUrl etc.) if we need them

Reply:
Done